### PR TITLE
Update docs, update cpu/memory request, limits for update-api agent DaemonSet

### PR DESCRIPTION
**Pull request by etungsten**
**Issue number:**
https://github.com/bottlerocket-os/bottlerocket-update-operator/issues/54
https://github.com/bottlerocket-os/bottlerocket-update-operator/issues/47

**Description of changes:**
```
Author: Erikson Tung <etung@amazon.com>
Date:   Mon Feb 15 13:17:29 2021 -0800

    deployment: change cpu and memory request, limits for api agent pods
    
    Significantly lower cpu, memory request/limits for the update-api agent
    DaemonSet


```

```
commit 96771ced4149e95ebc92ae65aeb63a436e573d74
Author: Erikson Tung <etung@amazon.com>
Date:   Mon Feb 15 13:05:04 2021 -0800

    README: change recommended update interface version to 2.0.0, add desc
    
    Update documentation to include short descriptions of the update
    interface versions. Include compatibility information

```

**Testing done:**

Monitored the memory usage of the agent using the update API. I never saw it peak above 25MB in physical memory usage.
Tried different CPU request values, the agent still works fine in updating nodes when requesting 10 millicores.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by bgdnlp** [comment]
Should the [example development deployment](https://github.com/bottlerocket-os/bottlerocket-update-operator/blob/develop/dev/deployment.yaml) that is linked to just below in the README also be adjusted or maybe removed?

To me it seems to be a bit out of date and the suggested deployment looks simple enough to serve as a starting point, I would completely remove that line.

**Review by jahkeup — APPROVED** [review]

**Review by srgothi92 — APPROVED** [review]

**Comment by etungsten** [comment]
Push above removes mention of the the development deployment as a customization starting point since that might be outdated.

Fixing it up should be addressed separately

File: README.md [hunk 1]
```diff
@@ -76,7 +76,7 @@ kubectl label node $(kubectl get nodes -o jsonpath='{.items[*].metadata.name}')
 
 Each workload resource may have additional constraints or scheduling affinities based on each node's labels in addition to the `bottlerocket.aws/updater-interface-version` label scheduling constraint.
 
-Customized deployments may use the [suggested deployment](./update-operator.yaml) or the [example development deployment](./dev/deployment.yaml) as a starting point, with customized container images specified if needed.
+Customized deployments may use the [suggested deployment](./update-operator.yaml) as a starting point, with customized container images specified if needed.
```
Comment (srgothi92) [inline README.md:79]: was this change planned to be part of another commit where you are updating the read me?

**Review by srgothi92 — APPROVED** [review]
Looks good! except for one minor confusion of commits I think. Multiple commit are pain to update especially if change is required in last commit in the chain. 

> Fixing it up should be addressed separately

Do you know where we use `./dev/deployment.yaml` ? Should we delete it instead of fixing it ? I feel we should delete `kind-cluster.yaml ` as well, we cannot use that as direct alternative to our actual bottlerocket cluster. It adds more confusion then to help. However, all of these suggestion are not for this PR, we can create an issue for cleanup.

File: README.md (hunk 1 above)
Comment (etungsten) [reply README.md:79]: Ah, missed that. I'll fix it.

Fixed below
